app/controllers/posts_controller.rb

```diff
@@ -50,7 +50,7 @@ def set_post
 
   def confirm_user
     set_post
-    if current_user.id != @postuser.id
+    if current_user.id != @post user.id
```

Comment:
ここ見落としてました🙇‍♂️
```suggestion
    if current_user.id != @post.user.id
```
が正しいですね！